src/menu/Menu.ts

```diff
@@ -0,0 +1,186 @@
+import { createHandle } from '@dojo/core/lang';
+import uuid from '@dojo/core/uuid';
+import { v } from '@dojo/widget-core/d';
+import { DNode } from '@dojo/widget-core/interfaces';
+import ThemeableMixin, { theme, ThemeableProperties } from '@dojo/widget-core/mixins/Themeable';
+import WidgetBase from '@dojo/widget-core/WidgetBase';
+import * as css from './styles/menu.m.css';
+import { Keys } from '../common/util';
+
+export type Orientation = 'horizontal' | 'vertical';
+
+export type Role = 'menu' | 'menubar';
+
+/**
+ * @type MenuProperties
+ *
+ * Properties that can be set on a Menu component.
+ *
+ * @property active             Determines whether the menu has focus.
+ * @property activeIndex        Determines the index of the focused item.
+ * @property disabled           Determines whether the menu is disabled.
+ * @property id                 The widget ID. Defaults to a random string.
+ * @property orientation        Determines whether the menu is rendered horizontally.
+ * @property role               The value to use for the menu's `role` property. Defaults to 'menu'.
+ */
+export interface MenuProperties extends ThemeableProperties {
+	active?: boolean;
+	activeIndex?: number;
+	disabled?: boolean;
+	id?: string;
+	orientation?: Orientation;
+	role?: Role;
+}
+
+export const enum Operation {
+	decrease,
+	increase
+}
+
+export const MenuBase = ThemeableMixin(WidgetBase);
+
+@theme(css)
+export class Menu extends MenuBase<MenuProperties> {
```

Comment:
We should probably have a custom element descriptor for all of these components.